doc/specification/language-proposal.md

```diff
@@ -0,0 +1,1518 @@
+
+Access to a field in a record that is bound to an identifier can be made via the corresponding qualified identifier (see section _Qualified Identifiers_); for records that are not bound to an identifier, the _record-field-expr_ provides the same function, also using the period as an field access operator.
+
+Records of different types can be combined with the ***preferential-union*** operator: given a record _r_ with fields _fᵢ_ and values _vᵢ_ for _i_∈_I_, and another record _s_ with fields _gⱼ_ and values _wⱼ_ for _j_∈_J_, the expression _r_ ⊔ _s_ is a record with a set of fields {_fᵢ_}∪{_gⱼ_}, and the value of a field _h_ in this record is _vᵢ_ if there exists a field _fᵢ_=_h_, else _wⱼ_ for the field _wⱼ_=_h_. Essentially, the record comprises all the fields of the left hand side, together with any fields from the record on the right hand side which don't have the same name as a field in the first record.
+
+
+### Precedence
+<a id="precedence"/>
+
+Where this is ambiguity in expression parsing, operators should be considered in the following order of decreasing precedence. In the table, _f_, _j_, _x_ and _y_ stand for expressions of the appropriate type.
+
+| Operator                   | Description              | Associativity | Notes |
+|----------------------------|--------------------------|---------------|-|
+| _x_`.`_y_                  | Record field             | left          | |
+| _f_`(`…`)`                 | Function application     | left          | [1](#precedence-note-1) |
+| _x_`^`_y_, _xʲ_            | Exponentiation           | right         | [2](#precedence-note-2) |
+| `-`                        | Unary minus              | right         | [3](#precedence-note-3) |
+| `√`                        | Square root              | right         | [3](#precedence-note-3) |
+| _x_`·`_y_, _x_`/`_y_       | Product, quotient        | left          | |
+| _x_`+`_y_, _x_`-`_y_       | Sum, difference          | left          | |
+| `⊔`                        | Preferential union       | left          | |
+| `<`, `≤`, `>`, `≥`         | Order comparitsons       | left          | |
+| `==`, `≠`                  | Compare equal, not equal | left          | |
+| `not`                      | Logical not              | right         | |
+| `and`                      | Logical and              | left          | |
+| `or`                       | Logical or               | left          | |
+| `:` _type-expr_            | Type assertion           | left          | |
+| `if` _x_ `then` y `else` z | Ternary conditional      | right         | |
+| `\|` _x_ → _y_             | Case expression          | right         | |
+| `let`, `with`              | Value bindings           | right         | |
+
+Notes:
+
+1. <a id="precedence-note-1"/> Function application can be considered left associative in accordance with mathematical convention, but it cannot arise while function values are unable to be returned from an Arblang function.
```

Comment:
Suppose we allowed functions to return functions and take functions as arguments. Then we could have an ambiguity in the expression `f(g)(x)` where `f` and `g` are functions. Should this be interpreted as *f*(*g*(*x*)) or (*f*(*g*))(*x*)?

But because of the restrictions on passing functions around, if `f(g)` is type valid, `g` cannot be a function.

Reply:
Ah, okay! I had misunderstood the statement. Thanks!